wagtail/wagtailcore/blocks/field_block.py

```diff
@@ -383,18 +384,22 @@ def clean(self, value):
 
 
 class PageChooserBlock(ChooserBlock):
+    from wagtail.wagtailcore.models import Page
+    page_model = Page
+
     def __init__(self, can_choose_root=False, **kwargs):
         self.can_choose_root = can_choose_root
         super(PageChooserBlock, self).__init__(**kwargs)
 
     @cached_property
     def target_model(self):
-        from wagtail.wagtailcore.models import Page  # TODO: allow limiting to specific page types
-        return Page
+        return self.page_model
 
     @cached_property
     def widget(self):
         from wagtail.wagtailadmin.widgets import AdminPageChooser
+        ct = ContentType.objects.get_for_model(self.page_model)
+        return AdminPageChooser(content_type=ct, can_choose_root=self.can_choose_root)
```

Comment:
As of 09878fcfd87f5b5ea67dd8d178667bce344c02e0, you can pass the page model straight in using the ``page_models`` attribute:

eg: ``AdminPageChooser(page_models=[self.page_model], can_choose_root=self.can_choose_root)``